server/neptune/workflows/internal/terraform/workflow.go

```diff
@@ -221,6 +221,26 @@ func (r *Runner) Apply(ctx workflow.Context, root *terraform.LocalRoot, serverUR
 func (r *Runner) Run(ctx workflow.Context) error {
 	r.MetricsHandler.Gauge(ActiveTerraformWorkflowStat).Update(1)
 	defer r.MetricsHandler.Gauge(ActiveTerraformWorkflowStat).Update(0)
+	var err error
+	// make sure we are updating state on completion.
+	defer func() {
+		reason := state.SuccessfulCompletionReason
+		if r := recover(); r != nil || err != nil {
```

Comment:
when will err not be nil

Reply:
what do you mean? err is set to the value of `run`?

Reply:
oops missed that